presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java

```diff
@@ -275,6 +281,15 @@ public Block getLoadedBlock()
         return lazyData.getFullyLoadedBlock();
     }
 
+    @Override
+    public Block loadSelectedPositions(SelectedPositions selectedPositions)
+    {
+        if (isLoaded()) {
+            return getBlock().loadSelectedPositions(selectedPositions);
+        }
+        return lazyData.getFullyLoadedBlock(selectedPositions);
```

Comment:
I'm curious why this call isn't lazy also.  All of the other calls like this delay computation until actual usage.  If you do decide to make this lazy, it should continue to prune the existing selected positions.

Reply:
I renamed `loadSelectedPositions` to be `getLoadedBlock` overload.